runtime/src/lib.rs

```diff
@@ -1041,30 +1044,87 @@ impl pallet_treasury::Config for Runtime {
     type BurnDestination = ();
     type Currency = Balances;
     type MaxApprovals = MaxApprovals;
-    type OnSlash = ();
+    type OnSlash = Treasury;
     type PalletId = TreasuryPalletId;
     type ProposalBond = ProposalBond;
-    type ProposalBondMaximum = ();
+    type ProposalBondMaximum = ProposalBondMaximum;
     type ProposalBondMinimum = ProposalBondMinimum;
     type RejectOrigin = EitherOfDiverse<
         EnsureRoot<AccountId>,
         pallet_collective::EnsureProportionMoreThan<AccountId, CouncilCollective, 1, 2>,
     >;
     type RuntimeEvent = RuntimeEvent;
     type SpendFunds = Bounties;
-    type SpendOrigin = frame_support::traits::NeverEnsureOrigin<u128>;
+    type SpendOrigin = EnsureWithSuccess<EnsureRoot<AccountId>, AccountId, MaxBalance>;
     type SpendPeriod = SpendPeriod;
     type AssetKind = ();
     type Beneficiary = AccountId;
-    type BeneficiaryLookup = Indices;
+    type BeneficiaryLookup = IdentityLookup<AccountId>;
     type BalanceConverter = UnityAssetBalanceConversion;
-    type Paymaster = PayFromAccount<Balances, TreasuryAccount>;
+    type Paymaster = PayFromTreasuryAccount;
     type PayoutPeriod = SpendPayoutPeriod;
     type WeightInfo = weights::pallet_treasury::WeightInfo<Runtime>;
     #[cfg(feature = "runtime-benchmarks")]
     type BenchmarkHelper = ();
 }
 
+pub struct PayFromTreasuryAccount;
+
+impl Pay for PayFromTreasuryAccount {
+    type Balance = Balance;
+    type Beneficiary = AccountId;
+    type AssetKind = ();
+    type Id = ();
+    type Error = DispatchError;
+
+    #[cfg(not(feature = "runtime-benchmarks"))]
+    fn pay(
+        who: &Self::Beneficiary,
+        _asset_kind: Self::AssetKind,
+        amount: Self::Balance,
+    ) -> Result<Self::Id, Self::Error> {
+        let _ = <Balances as fungible::Mutate<_>>::transfer(
+            &TreasuryAccount::get(),
+            who,
+            amount,
+            Preservation::Expendable,
+        )?;
+        Ok(())
+    }
+
+    #[cfg(feature = "runtime-benchmarks")]
+    fn pay(
+        who: &Self::Beneficiary,
+        _asset_kind: Self::AssetKind,
+        amount: Self::Balance,
+    ) -> Result<Self::Id, Self::Error> {
+        // In case of benchmarks, we adjust the value by multiplying it by 1_000_000_000_000, otherwise it fails with BelowMinimum limit error, because
+        // treasury benchmarks uses only 100 as the amount.
+        let _ = <Balances as fungible::Mutate<_>>::transfer(
+            &TreasuryAccount::get(),
+            who,
+            amount * 1_000_000_000_000,
```

Comment:
So I'm not gonna dig into the logic here, just gonna assume this is fine. But I'd at least change this to something like:

```suggestion
            amount * 100 * DOLLARS
```